### PR TITLE
feat(hepa-uv): Add CAN messages to control the Hepa/UV filter.

### PR DESCRIPTION
**Pull request by vegano1**
# Overview

We need to control the Hepa/UV filter from the Flex for testing and eventual system integration, this PR implements the CAN messages required to make this happen. It also adds a new method to query the time remaining before a software timer expires. This PR coincides with [Opentrons/opentrons#14452](https://github.com/Opentrons/opentrons/pull/14452) PR.

Closes: [RET-1422](https://opentrons.atlassian.net/browse/RET-1422) [RET-1423](https://opentrons.atlassian.net/browse/RET-1423)

# Test Plan

- [x] Make sure the Hepa fan still turns on/off with the push button
- [x] Make sure the UV Ballast still turns on/off with the push button when the Flex door is closed and reed switch is set.
- [x] Make sure that we can set/get the Hepa fan state and duty cycle with the SetHepaFanStateRequest and GetHepaFanStateRequest messages.
- [x] Make sure that we can set/get the UV light state and timeout with the SetHepaUVStateRequest and GetHepaUVStateRequest messages.
- [x] Make sure that the UV light ONLY turns on via CAN message when the door is closed and reed switch is set.
- [x] Make sure that the UV light turns OFF when the door is opened
- [x] After turning ON the UV light via a CAN message, make sure we can turn off the light when the push button is pressed.

# Change Log

- Added implementation for CAN messages to set/get Hepa fan state/duty cycle
- Added implementation for CAN messages to set/get UV Light state/timeout in seconds
- Added a get_remaining_time method to the FreeRTOSTimer class to know the time in ms before the timer expires.
- Integrated the new CAN messages into their corresponding Hepa Task and UV Task
- Refactored the Hepa Task logic so it takes into account the CAN messages so we aren't duplicating code.
- Refactored the UV Task logic so it takes into account the CAN messages so we aren't duplicating code.

# Review Requests

- Does anything not make sense with the can messages?

# Risk Assessment

Low, unreleased

[RET-1422]: https://opentrons.atlassian.net/browse/RET-1422?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RET-1423]: https://opentrons.atlassian.net/browse/RET-1423?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: cpp-utils/include/ot_utils/freertos/freertos_timer.hpp [hunk 1]
```diff
@@ -6,6 +6,9 @@
 #include "task.h"
 #include "timers.h"
 
+#define ROUNDED_DIV(A, B) (((A) + ((B) / 2)) / (B))
+#define PASSED_MS(TICKS) (uint32_t)ROUNDED_DIV((TICKS) * 1000, configTICK_RATE_HZ)
```
Comment (sfoster1) [inline cpp-utils/include/ot_utils/freertos/freertos_timer.hpp:10]: can we name this like `TICKS_TO_MS` or something, it's a little unclear just from the name what it does.

**Review by sfoster1 — APPROVED** [review]
Looks good, but let's change the name of that macro. Also, I think this functionality is getting to the point where it should really have tests.

**Comment by vegano1** [comment]
> Looks good, but let's change the name of that macro. Also, I think this functionality is getting to the point where it should really have tests.

agreed, I will add a test suite in up coming PR.

**Review by caila-marashaj — APPROVED** [review]
Awesome!

**Pull request merged**